File: python/program/BarangPremium.py

# Class Level 2: BarangPremium
from BarangElektronik import BarangElektronik

class BarangPremium(BarangElektronik):
    def __init__(self, id, nama, merek, harga, material, fitur_eksklusif):
        super().__init__(id, nama, merek, harga)
        self.material = material
        self.fitur_eksklusif = fitur_eksklusif

    def get_material(self):
        return self.material
    def get_fitur_eksklusif(self):
        return self.fitur_eksklusif

    def set_material(self, material):
        self.material = material
    def set_fitur_eksklusif(self, fitur_eksklusif):
        self.fitur_eksklusif = fitur_eksklusif

    def display(self, col_widths):
        return f"{super().display(col_widths)} {self.get_material():<{col_widths['Material']}} | {self.get_fitur_eksklusif():<{col_widths['Fitur Eksklusif']}} |"

File: java/program/BarangPremium.java

import java.util.Map;
import java.text.DecimalFormat;
// Class Level 2: BarangPremium
class BarangPremium extends BarangElektronik {
    protected String material;
    protected String fiturEksklusif;

    public BarangPremium(int id, String nama, String merek, double harga, String material, String fiturEksklusif) {
        super(id, nama, merek, harga);
        this.material = material;
        this.fiturEksklusif = fiturEksklusif;
    }

    // Getters
    public String getMaterial() { return material; }
    public String getFiturEksklusif() { return fiturEksklusif; }

    // Setters
    public void setMaterial(String material) { this.material = material; }
    public void setFiturEksklusif(String fiturEksklusif) { this.fiturEksklusif = fiturEksklusif; }

    // fungsi di override biar display ga tumpang tindih
    @Override
    public void display(Map<String, Integer> colWidths) {
        super.display(colWidths);
        System.out.printf(" %-" + colWidths.get("Material") + "s | %-" + colWidths.get("Fitur Eksklusif") + "s |", getMaterial(), getFiturEksklusif());
    }
}
